src/cpp/include/openvino/genai/tokenizer.hpp

```diff
@@ -25,6 +25,9 @@ struct TokenizedInputs {
     std::optional<ov::Tensor> token_type_ids;
 };
 
+class TokenizerImpl;
+class StructuredOutputController;
```

Comment:
Move `class TokenizerImpl;` to `private` where it was.

`class StructuredOutputController;` is now constructed from `TokenizerImpl` (great idea!). `StructuredOutputConfig` should be a `friend` anymore